lib/kramdown/converter/pdf.rb

```diff
@@ -608,6 +603,14 @@ def block_hash(opts)
         hash
       end
 
+      def open_file(base_dirs, path_or_url)
+        open(path_or_url)
+      rescue
+        next_dir = base_dirs.shift
+        raise "#{path_or_url} cannot be opened" unless next_dir
+
+        open_file(base_dirs, File.join(next_dir, path_or_url))
```

Comment:
Don't use recursion, not necessary here, use the `retry` statement as in the original code. Generally, you can just put the code from lines 156-161 in here.

Reply:
@gettalong the recursion is necessary here because itis trying the file name as given in `src` first, and then trying with base_dirs. Having the code as in 156-161 would not have the same effect (tried and figured that out).